providers.go

```diff
@@ -60,7 +63,7 @@ func file(s *Store, filename string, refresh bool, fn func([]byte) ([]Item, erro
 		return
 	}
 	inmem := inMemoryProvider(s, providername)
-	logrus.Infof("Configuration from file: %s", filename)
+	LogInfoFunc("configuration from file: %s", filename)
```

Comment:
As variable is overridable by anybody, I would protect this path of code
```suggestion
	if LogInfoFunc != nil {
		LogInfoFunc("configuration from file: %s", filename)
	}
```

Reply:
done